Makefile

```diff
@@ -13,6 +13,11 @@ CFLAGS += -D ENABLE_SDL
 CFLAGS += `sdl2-config --cflags`
 LDFLAGS += `sdl2-config --libs`
 
+# Whether to enable computed goto in riscv.c
+ifeq ("$(NO_COMPUTED_GOTO)", "")
+riscv.o: CFLAGS += -fno-gcse -fno-crossjumping -D ENABLE_COMPUTED_GOTO
```

Comment:
You should detect the compiler. If it is gcc, apply the options. Otherwise, omit.